libtenzir/builtins/operators/print.cpp

```diff
@@ -0,0 +1,217 @@
+      }
+      TENZIR_ASSERT(printer);
+      printer_ = printer->parse_printer(p);
+      TENZIR_ASSERT(printer_);
+      if (not printer_->prints_utf8()) {
+        diagnostic::error("print operator does not support binary formats")
+          .primary(printer_name_->source)
+          .throw_();
+      }
+    } catch (diagnostic& d) {
+      std::move(d)
+        .modify()
+        .usage("print <input> <printer> <args>...")
+        .docs("https://docs.tenzir.com/operators/print")
+        .throw_();
+    }
+  }
+
+  auto
+  operator()(generator<table_slice> input, operator_control_plane& ctrl) const
+    -> generator<table_slice> {
+    for (auto&& slice : input) {
+      if (slice.rows() == 0) {
+        co_yield {};
+        continue;
+      }
+      auto target_index
+        = slice.schema().resolve_key_or_concept_once(input_.inner);
+      if (not target_index) {
+        diagnostic::error("could not resolve `{}` for schema `{}`",
+                          input_.inner, slice.schema())
+          .primary(input_.source)
+          .throw_();
+      }
+      auto transform = [&](struct record_type::field field,
+                           std::shared_ptr<arrow::Array> array)
+        -> std::vector<
+          std::pair<struct record_type::field, std::shared_ptr<arrow::Array>>> {
+        if (not caf::holds_alternative<record_type>(field.type)) {
+          diagnostic::error("field {} is not of type record", field.name)
+            .primary(input_.source)
+            .throw_();
+        }
+        field.type = type{
+          fmt::format("{}.{}", slice.schema().name(),
+                      caf::get<record_type>(slice.schema()).key(*target_index)),
+          field.type,
+        };
+        auto rb = arrow::RecordBatch::Make(
+          field.type.to_arrow_schema(), array->length(),
+          static_cast<const arrow::StructArray&>(*array).Flatten().ValueOrDie());
+        auto slice = table_slice{rb, field.type};
+        auto builder = series_builder{type{string_type{}}};
+        for (size_t i = 0; i < slice.rows(); i++) {
+          auto row = subslice(slice, i, i + 1);
+          auto chunks = std::vector<chunk_ptr>{};
+          try {
+            auto printer_instance = printer_->instantiate(field.type, ctrl);
+            std::ranges::copy(printer_instance->get()->process(row),
+                              std::back_inserter(chunks));
+            std::ranges::copy(printer_instance->get()->finish(),
+                              std::back_inserter(chunks));
+            std::erase_if(chunks, [](const chunk_ptr& chunk) {
+              return not chunk or chunk->size() == 0;
+            });
+          } catch (diagnostic diag) {
+            std::move(diag)
+              .modify()
+              .severity(severity::warning)
+              .emit(ctrl.diagnostics());
+            builder.null();
+            continue;
+          } catch (...) {
+            std::rethrow_exception(std::current_exception());
```

Comment:
```suggestion
            throw;
```

👆 Absolute nitpick, but that's more idiomatic (and there are subtle differences). Can also ignore ^^

Reply:
TIL, I didn't know this was a thing.

Reply:
Although I just noticed that you could also just omit the entire `catch (...)` clause then... 😅  

Well, whatever, I won't block CI once more just because of that.